### PR TITLE
Add HTML file support & more WiX file support to LocProject.json generation

### LINKED ISSUE #12090: Pick up HTML files in LocProject.json generation
- [x] This issue is blocking 
- [ ] This issue is causing unreasonable pain

Installer needs html files to be localized so we're gonna throw those in the template real quickly.

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements
### Release Note Description
Do not include in release notes.

### PR DESCRIPTION
**Pull request by jonfortescue**
Resolves #12090.

This adds code to the `generate-locproject.ps1` script to pick up macOS HTML files. Specifically, it finds the English version of these files and then copies it to the parent directory to act as a neutral language file for translators (which is the appropriate OneLocBuild process). Additionally, I noticed that some repos did not have neutral language wxl files checked in for their WiX files (like aspnetcore does which was the basis for this feature). So this also adds functionality to handle that.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: eng/common/generate-locproject.ps1 [hunk 1]
```diff
@@ -34,6 +34,25 @@ $jsonTemplateFiles | ForEach-Object {
 $jsonWinformsTemplateFiles = Get-ChildItem -Recurse -Path "$SourcesDirectory" | Where-Object { $_.FullName -Match "en\\strings\.json" } # current winforms pattern
 
 $wxlFiles = Get-ChildItem -Recurse -Path "$SourcesDirectory" | Where-Object { $_.FullName -Match "\\.+\.wxl" -And -Not( $_.Directory.Name -Match "\d{4}" ) } # localized files live in four digit lang ID directories; this excludes them
+if (-not $wxlFiles) {
+    $wxlEnFiles = Get-ChildItem -Recurse -Path "$SourcesDirectory" | Where-Object { $_.FullName -Match "\\1033\\.+\.wxl" } # pick up en files specifically
```
Comment (dougbu) [inline eng/common/generate-locproject.ps1:38]: This nice addition tempts me to remove the .wxl En files in parent directories once we update to a version of Arcade containing this. The duplication has been somewhat annoying.

**Review by dougbu — DISMISSED** [review]
Looks good to me. Without testing in the CI, can this change be verified❔ E.g. should @marcpopMSFT duplicate this change in an installer PR and run the loc pipeline against that branch❔

File: eng/common/generate-locproject.ps1 (hunk 1 above)
Comment (MattGal) [inline eng/common/generate-locproject.ps1:38]: Probably merits a comment about 1033 = EN-US and why you need the EN files.

**Review by MattGal — DISMISSED** [review]

**Review by garath — APPROVED** [review]

**Review by MattGal — APPROVED** [review]

**Comment by jonfortescue** [comment]
@dougbu sorry, just seeing your comment about repro now. I've actually run this locally against the installer repo and it generated a locproject.json with the WiX and macos lproj files included!

**Review by MattGal — APPROVED** [review]

**Pull request merged**